jena-text/src/main/java/org/apache/jena/query/text/TextDatasetFactory.java

```diff
@@ -191,5 +213,50 @@ public static DatasetGraph createSolrIndex(DatasetGraph base, SolrServer server,
         TextIndex index = createSolrIndex(server, entMap) ;
         return create(base, index, true) ; 
     }
+ 
+    protected static class LuceneIndexMap {
+    	
+    	private static HashMap<Object, WeakReference<TextIndex>> map = new HashMap<Object, WeakReference<TextIndex>>();
+
+    	protected synchronized TextIndex get(Directory directory) {
```

Comment:
Yes, there is some commonality.  It may be possible to factor something out from both.

There are differences.  One uses a WeakHashMap and the other a normal HashMap and the handling of keys is a little different.

I reckoned the common functionality was not worth factoring out.  Ymmv.